src/io/DumpH5MDParallel.cpp

```diff
@@ -0,0 +1,322 @@
+/*
+  This file is part of ESPResSo++.
+
+  ESPResSo++ is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "DumpH5MDParallel.hpp"
+
+#include "bc/BC.hpp"
+#include "iterator/CellListIterator.hpp"
+#include "storage/Storage.hpp"
+#include "Version.hpp"
+
+namespace espressopp {
+namespace io {
+
+template <typename T>
+void DumpH5MDParallel::writeParallel(hid_t fileId, const std::string& name,
+                                     const std::vector<hsize_t>& globalDims,
+                                     const std::vector<hsize_t>& localDims,
+                                     const std::vector<T>& data) {
+    CHECK_EQUAL(globalDims.size(), localDims.size());
+    CHECK_EQUAL(data.size(),
+                std::accumulate(localDims.begin(), localDims.end(), 1, std::multiplies<hsize_t>()));
+
+    auto dataspace = CHECK_HDF5(H5Screate_simple(globalDims.size(), globalDims.data(), nullptr));
+    auto dataset = CHECK_HDF5(H5Dcreate(fileId, name.c_str(), typeToHDF5<T>(), dataspace, H5P_DEFAULT,
+                             H5P_DEFAULT, H5P_DEFAULT));
+
+    std::vector<hsize_t> offset(globalDims.size(), 0);
+    offset[0] = particleOffset;
+    std::vector<hsize_t> stride(globalDims.size(), 1);
+    std::vector<hsize_t> count(globalDims.size(), 1);
+    for (auto i = 0; i < globalDims.size(); ++i) {
+        CHECK_LESS_EQUAL(localDims[i] + offset[i], globalDims[i], "i = " << i);
+    }
+    auto dstSpace = CHECK_HDF5(H5Dget_space(dataset));
+    CHECK_HDF5(H5Sselect_hyperslab(dstSpace, H5S_SELECT_SET, offset.data(), stride.data(),
+                                   count.data(), localDims.data()));
+
+    std::vector<hsize_t> localOffset(globalDims.size(), 0);
+    auto srcSpace = CHECK_HDF5(H5Screate_simple(localDims.size(), localDims.data(), NULL));
+    CHECK_HDF5(H5Sselect_hyperslab(srcSpace, H5S_SELECT_SET, localOffset.data(), stride.data(),
+                                   count.data(), localDims.data()));
+
+    auto datawrite_id = CHECK_HDF5(H5Pcreate(H5P_DATASET_XFER));
```

Comment:
Are there any particular reason to use snake case?

Reply:
no, corrected.